index.dd

```diff
@@ -240,6 +240,30 @@ void main()
 }
 ----
 )
+$(EXTRA_EXAMPLE_RUNNABLE Subtyping with alias this,
+----
+struct Point
+{
+    private double[2] p;
+    // If Point.<function> isn't defined, use p.<function>
+    alias p this;
+    double dot(Point rhs)
+    {
+        return p[0] * rhs.p[0] + p[1] * rhs.p[1];
+    }
+}
+void main()
+{
+    import std.stdio : writeln;
+    Point p1, p2;
+    // Point behaves like a double[2]
+    p1 = [2, 1], p2 = [1, 1];
+    assert(p1[$ - 1] == 1);
+    // but with extended functionality
```

Comment:
I would prefer ellipses to show continuity:

```D
// `Point` behaves like `double[2]`...
p1 = [2, 1], p2 = [1, 1];
assert(p1[$ - 1] == 1);
// ...but with extended functionality
```

And you can tell I prefer tick marks to disambiguate code from English even in the comments.